### DISCUSSION TITLE
how the swarm node network aggregate the model parameter?

### DISCUSSION THREAD

**Discussion by EdwinH666** [HewlettPackard/swarm-learning#45] [General]
In the calculation of node aggregation, do the model parameters of the lead node participate in the aggregation sharing calculation? 
How do each node's local model update and iteration in local epoch? Is gradient information used?

**Comment by RadhakrishnaJ** [comment]
Leader weights also participates in merge algorithm to produce merged weights. Merged weights are shared with all nodes.

Swarm merging happens when sync interval is met. Epoch divided into batches, once number of batches trained matches with sync interval then Swarm network runs to process merging of weights. Merged weights are loaded into local model and then continues with training process. This process continues through out the training when epochs are training.

Gradient information is used to generate local weights as per the training process.

**Reply by EdwinH666** [reply]
Hello, I would like to ask whether the aggregation algorithm of group nodes when sharing node model information is weighted average? Compared to the FedAVG algorithm, does the leader node serve as the central server? Theoretically, is the performance consistent with that of the FedAVG algorithm?
